qiime2_pipeline.bash

```diff
@@ -0,0 +1,177 @@
+  --i-demultiplexed-seqs "${PROJECT_DIR}/demux.qza" \
+  --p-trim-left-f 0 \
+  --p-trunc-len-f 230 \
+  --p-trim-left-r 0 \
+  --p-trunc-len-r 230 \
+  --p-n-threads 8 \
+  --o-representative-sequences "${PROJECT_DIR}/rep-seqs.qza" \
+  --o-table "${PROJECT_DIR}/table.qza"
+
+qiime feature-table summarize \
+  --i-table "${PROJECT_DIR}/table.qza" \
+  --o-visualization "${PROJECT_DIR}/table.qzv" \
+  --m-sample-metadata-file ${MAPPING_FP}
+
+qiime feature-table tabulate-seqs \
+  --i-data "${PROJECT_DIR}/rep-seqs.qza" \
+  --o-visualization "${PROJECT_DIR}/rep-seqs.qzv"
+
+qiime tools export \
+  "${PROJECT_DIR}/table.qzv" \
+  --output-dir "${PROJECT_DIR}/table"
+
+qiime tools export \
+  "${PROJECT_DIR}/table.qza" \
+  --output-dir "${PROJECT_DIR}/table"
+
+###=====================
+###  TAXONOMIC ANALYSIS
+###=====================
+
+qiime feature-classifier classify-sklearn \
+  --i-classifier ${CLASSIFIER_FP} \
+  --i-reads "${PROJECT_DIR}/rep-seqs.qza" \
+  --o-classification "${PROJECT_DIR}/taxonomy.qza"
+
+qiime metadata tabulate \
+  --m-input-file "${PROJECT_DIR}/taxonomy.qza" \
+  --o-visualization "${PROJECT_DIR}/taxonomy.qzv"
+
+qiime tools export \
+  "${PROJECT_DIR}/taxonomy.qza" \
+  --output-dir "${PROJECT_DIR}/taxonomy"
+
+###=====================
+###  GENERATE TREES
+###=====================
+
+qiime alignment mafft \
+  --i-sequences "${PROJECT_DIR}/rep-seqs.qza" \
+  --o-alignment "${PROJECT_DIR}/aligned-rep-seqs.qza"
+
+qiime alignment mask \
+  --i-alignment "${PROJECT_DIR}/aligned-rep-seqs.qza" \
+  --o-masked-alignment "${PROJECT_DIR}/masked-aligned-rep-seqs.qza"
+
+qiime phylogeny fasttree \
+  --i-alignment "${PROJECT_DIR}/masked-aligned-rep-seqs.qza" \
+  --o-tree "${PROJECT_DIR}/unrooted-tree.qza"
+
+qiime phylogeny midpoint-root \
+  --i-tree "${PROJECT_DIR}/unrooted-tree.qza" \
+  --o-rooted-tree "${PROJECT_DIR}/rooted-tree.qza"
+
+###=====================
+###  ALPHA AND BETA DIVERSITY
+###=====================
+
+qiime diversity core-metrics \
+  --i-phylogeny "${PROJECT_DIR}/rooted-tree.qza" \
+  --i-table "${PROJECT_DIR}/table.qza" \
+  --p-sampling-depth 10000 \
+  --output-dir "${PROJECT_DIR}/core-metrics-results"
+
+qiime tools export \
+  "${PROJECT_DIR}/core-metrics-results/observed_otus_vector.qza" \
+  --output-dir "${PROJECT_DIR}/core-metrics-results/observed_otu"
+
+qiime tools export \
+  "${PROJECT_DIR}/core-metrics-results/shannon_vector.qza" \
+  --output-dir "${PROJECT_DIR}/core-metrics-results/shannon"
+
+qiime tools export \
+  "${PROJECT_DIR}/core-metrics-results/faith_pd_vector.qza" \
+  --output-dir "${PROJECT_DIR}/core-metrics-results/faith"
+
+qiime tools export \
+  "${PROJECT_DIR}/core-metrics-results/unweighted_unifrac_distance_matrix.qza" \
+  --output-dir "${PROJECT_DIR}/core-metrics-results/uu"
+
+qiime tools export \
+  "${PROJECT_DIR}/core-metrics-results/weighted_unifrac_distance_matrix.qza" \
+  --output-dir "${PROJECT_DIR}/core-metrics-results/wu"
+
```

Comment:
Suggest generating a PCoA visualization, excluding the positive and negative control samples.